nuxeo-drive-client/nxdrive/tests/test_local_changes_when_offline.py

```diff
@@ -0,0 +1,146 @@
+'''
+Created on 15-Dec-2016
+
+    Test if changes made to local file system when nuxeo-drive is offline 
+    sync's back later when nuxeo-drive becomes online
+
+@author: dgraja
+@author: mkeshava
+
+'''
+from common_unit_test import UnitTestCase
+from nxdrive.tests.common_unit_test import FILE_CONTENT
+from nxdrive.osi import AbstractOSIntegration
+from nxdrive.logging_config import get_logger
+import shutil
+
+log = get_logger(__name__)
+if AbstractOSIntegration.is_windows():
+    from win32com.shell import shell, shellcon
+else:
+    import xattr
+    import Cocoa
+
+
+class TestOfflineChangesSync(UnitTestCase):
+    '''
+        All changes made in local PC while drive is offline should sync later
+        when drive comes back online
+        offline can be one of: suspended, exited or disconnect
+    '''
+    def setUp(self):
+        self.engine_1.start()
+        log.debug("Configure drive and wait for sync")
+        self.wait_sync(wait_for_async=True)
+
+        self.localpc = self.local_client_1
+        self.server = self.remote_document_client_1
+        # Create a folder and a file in server side.
+        # Wait for Drive to finish sync (download)
+        log.debug("Server: Create folder and upload a file")
+        self.folder1_remote = self.server.make_folder("/", "Folder1")
+        self.file1_remote = self.server.make_file(self.folder1_remote, "File1.txt", FILE_CONTENT)
+        self.wait_sync(wait_for_async=True)
+
+        # Verify that the folder and file are sync'd (download) successfully
+        log.debug("Client: Verify sync successful")
+        self.assertTrue(self.localpc.exists("/Folder1/File1.txt"),"File synced")
+
+    def copy_with_xattr(self, src, dst):
+        """
+            custom copy tree method that also copies xattr along with shutil.copytree functionality
+        """
+        if AbstractOSIntegration.is_windows():
+            # Make a copy of file1 using shell (to copy including xattr)
+            shell.SHFileOperation((0, shellcon.FO_COPY, 
+                                   src, dst, shellcon.FOF_NOCONFIRMATION, None, None))
+        elif AbstractOSIntegration.is_mac():
+            self.fm = Cocoa.NSFileManager.defaultManager()
+            self.fm.copyItemAtPath_toPath_error_(src, dst, None)
+        else:
+            shutil.copy2(src, dst)
+            remote_id = unicode(xattr.getxattr(src, "ndrive"), 'utf-8')
```

Comment:
On GNU/Linux, you had to prefix the key with `user.`, see [local_client.py#L406](https://github.com/nuxeo/nuxeo-drive/blob/817b335431e2422856f06567add6716244945a42/nuxeo-drive-client/nxdrive/client/local_client.py#L406).